wiki/python/DEVELOPER_GUIDE.md

```diff
@@ -36,6 +36,20 @@ except CudaRuntimeError as cre:
     print("Cuda Error! '%s'" % str(cre))
 ```
 
+## Dependencies
+The cuML python package has the following dependencies:
+<list of dependencies or instructions how to list them>
```

Comment:
You can find the dependencies easily using the following conda command: 

```bash
conda search -c rapidsai-nightly cuml --info
```

Which will give a relatively large list of packages found (since it is looking at the nightlies), but the last entry _should_ be enough, for example right now it is: 

```bash
  - cudatoolkit >=11.0.221,<11.1.0a0
  - cudf 0.17.*
  - cupy >7.1.0,<9.0.0a0
  - dask >=2.12.0
  - dask-cudf 0.17.*
  - distributed >=2.12.0
  - joblib >=0.11
  - libcuml 0.17.0a201017.*
  - libcumlprims 0.17.*
  - nccl >=2.5
  - python >=3.8,<3.9.0a0
  - python_abi 3.8.* *_cp38
  - treelite 0.93.*
  - ucx-py 0.17.*
```

Otherwise the dependencies are in: https://github.com/rapidsai/cuml/blob/branch-0.17/conda/recipes/cuml/meta.yaml

Also, dependencies of our rapids dependencies usually don't need to be guarded, for example `numba` is not there, but since cuDF needs and pins it, it is safe to import directly.